### PR TITLE
Improve engine repo readme

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,19 +29,28 @@
...
+Cocos Creator inherited many good qualities and cool features from its previous versions, such as high performance low level C++ implementation, intuitive editor, cross-platform support. It supports native platforms, web platforms and rapidly expanding instant gaming platforms, including Windows, Mac, iOS, Android, HarmonyOS, Web, Facebook Instant Games, WeChat Mini Game and TikTok Mini Games.
 
-![image](https://user-images.githubusercontent.com/1503156/111037166-f27c7600-845d-11eb-988f-4c2c8b5c7321.png)
+Furthermore, Cocos Creator has pushed our technology to a whole new level for high performance with scalability on various platforms, full extensibility and easy development.
+
+1. **Modern Graphics**: Our GFX implementation is designed to adapt to the modern graphics APIs, it uses Vulkan on Windows and Android, Metal on Mac OS and iOS, WebGL on Web platform.
```
Comment (jareguo) [inline README.md]: I think we should weaken the use of "Our" to represent our open source attitude, it is a work that we want to build and own together with everyone, not just "ours". So replacing all the "our" in this readme with "the" (or sth else) would probably be more like an open source product.

File: README.md [hunk 2]
```diff
@@ -29,19 +29,28 @@
...
+2. **High Performance**: The runtime engine is built with half C++ and half TypeScript, low level infrastructure, native platform adaptation, renderer and scene management are all written in C++ to ensure high runtime performance. We continue to move heavy lifting work to native as much as possible.
+3. **Customizable Render Pipeline**: The render pipeline is designed to be fully customizable, it has supported the builtin forward and deferred render pipeline across all platforms. Developers can customize their own render pipeline following the same approach.
+4. **Extensible Surface Shader**: The material system is built on our own effect format which uses GLSL 300, the shader programs will be converted to suitable runtime format automatically. The surface shader permit to fully customize the surface material while ensuring universal lighting model.
```
Comment (jareguo) [inline README.md]: Similarly, "our own" -> Cocos Effect Format ?

File: README.md [hunk 3]
```diff
@@ -29,19 +29,28 @@
...
+5. **Physically Based Rendering**: Our standard effect adopts physically based rendering, along with the physically based camera and the lighting based on physical metrics, developers can easily achieve realistic and seamless rendering results across different environment.
```
Comment (jareguo) [inline README.md]: Physically Based Rendering -> PBR for more SEO firendly

File: README.md [hunk 4]
```diff
@@ -72,16 +81,16 @@ Please refer to [native readme](native/README.md) if want to develop native appl
...
 - [Test Cases](https://github.com/cocos/cocos-test-projects): Unit test scenes for every engine module.
+- [Example Cases](https://github.com/cocos/cocos-example-projects): Simple yet expressive demo scenes for baseline testing and topic-specific case study.
+- [Awesome Cocos](https://github.com/cocos/awesome-cocos): You can find out other useful tools and show cases here.
 
 ## Links
 
-- [Official site](https://www.cocos.com/products#CocosCreator)
-- [Download](https://www.cocos.com/creator)
+- [Official site](https://www.cocos.com/)
+- [Download](https://www.cocos.com/creator/download)
```
Comment (jareguo) [inline README.md]: For historical reasons, the current English website is not this link. It's https://www.cocos.com/en/creator/download in my browser.

File: README.md [hunk 5]
```diff
@@ -72,16 +81,16 @@ Please refer to [native readme](native/README.md) if want to develop native appl
...
 - [Documentation](https://docs.cocos.com/creator/manual/en/)
 - [API References](https://docs.cocos.com/creator/api/en/)
 - [Forum](https://discuss.cocos2d-x.org/c/creator)
-- [Road Map](https://trello.com/b/JWVRRxMG/cocos-creator-roadmap)
+- [Projects and road map](https://github.com/orgs/cocos/projects?query=is%3Aopen&type=new)
```
Comment (jareguo) [inline README.md:122]: There's so little content in Projects and it's actually hard for users to find the key updates. So I think trello could continue to be used.

File: README.md (hunk 1 above)
Comment (pandamicro) [reply README.md]: 👍

File: README.md (hunk 5 above)
Comment (pandamicro) [reply README.md:122]: Trello is not tracked and regularly updated, we should do the effort to use Projects, and it's been gradually adopted, it's also much more reasonable with our open issues.

**Review by jareguo — APPROVED** [review]

**Comment by pandamicro** [comment]
https://github.com/cocos/cocos-engine/issues/10610

**Pull request merged**